### PR TITLE
Finished code_6000.c

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

File: src/code_6000.c [hunk 1]
```diff
@@ -50,16 +48,25 @@ s32 heap_free(s32 size) {
...
+s32 collision_heap_malloc(s32 size) {
+    if (!(*gGameStatusPtr)->isBattle) {
+        return _heap_malloc(&D_80268000, size);
+    } else {
+        return _heap_malloc(&D_803DA800, size);
+    }
+}
 
-#ifdef NON_MATCHING
 s32 collision_heap_free(s32 size) {
```
Comment (pixel-stuck) [inline src/code_6000.c:66]: Is this really size? Looking up a heap entry just via size seems uhhh, well, suspect to say the least. Usually free takes in a  pointer and not a size. (Also seems like _heap_free can handle multiple heaps so you need to tell it which one, hence the first ptr). I'd make the arg to this func take in a ``void *`` to a var called `buffer` or `buf` or so.

Comment (ethteck) [reply src/code_6000.c:66]: Oh yeah oops it's a pointer, not a size. clover calls it "data" which makes sense to me. idk

Comment (pixel-stuck) [reply src/code_6000.c:66]: Data sounds like a fine name to me

File: src/code_42e0_len_1f60.c [hunk 2]
```diff
@@ -10,7 +10,7 @@ INCLUDE_ASM("code_42e0_len_1f60", func_80028FE0);
 
 INCLUDE_API_ASM("code_42e0_len_1f60", length2D);
 
-HeapNode* INCLUDE_ASM("code_42e0_len_1f60", _heap_create, s32 addr, s32 size);
+HeapNode* INCLUDE_ASM("code_42e0_len_1f60", _heap_create, u32* addr, s32 size);
```
Comment (nanaian) [inline src/code_42e0_len_1f60.c:13]: Perhaps this should be `void*`

Comment (ethteck) [reply src/code_42e0_len_1f60.c]: yee fixed.

File: src/code_6000.c [hunk 3]
```diff
@@ -23,7 +21,7 @@ s32 general_heap_free(s32 size) {
 }
 
 s32 battle_heap_create(void) {
-    if (_heap_create(&D_803DA800, 0x25800) == -1) {
+    if ((s32)_heap_create(&D_803DA800, 0x25800) == -1) {
```
Comment (pixel-stuck) [inline src/code_6000.c:24]: why the s32 cast?

File: src/code_6000.c [hunk 4]
```diff
@@ -50,16 +48,25 @@ s32 heap_free(s32 size) {
...
+s32 collision_heap_malloc(s32 size) {
+    if (!(*gGameStatusPtr)->isBattle) {
+        return _heap_malloc(&D_80268000, size);
+    } else {
+        return _heap_malloc(&D_803DA800, size);
+    }
+}
 
-    return _heap_free((isBattle == 0) ? (&D_80268000) : (&D_803DA800), size);
+s32 collision_heap_free(UNK_PTR data) {
```
Comment (pixel-stuck) [inline src/code_6000.c:66]: imo, this shouldn't be UNK_PTR, it should be a generic pointer type (void*), which in C can take in any kind of pointer and thus is used for free, memcpy, etc.

**Pull request merged**